### PR TITLE
chore: move mrc voting to allo sdk

### PR DESCRIPTION
**Pull request by vacekj**
Fixes: #2817 

## Description

moves the donation part and related types of voting using MRC on Allo v1 to the allo sdk.

## Testing

donating to v1 rounds on various chains and using various tokens - dai, erc20, native

## Checklist

This PR:

- [x] Includes only changes relevant to the original ticket. Significant refactoring needs to be separated.
- [x] Doesn't disable eslint rules.
- [x] Doesn't work around the type checker (including but not limited to: type casts, non-null assertions, `@ts-ignore`, unjustified optional values).
- [x] Doesn't contain commented out code.
- [x] If adding/updating a feature, it adds/updates its test script on Notion.

### CODE REVIEW

**Comment by gravityblast** [comment]
I tried the checkout with one project in the cart from this test round and it fails:

Screenshot 2024-02-22 at 16 26 49

**Review by gravityblast — CHANGES_REQUESTED** [review]
added a comment above with an error

File: packages/common/src/allo/addresses/mrc.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  [ChainId.DEV2]: "0x0000000000000000000000000000000000000000",
+  [ChainId.MAINNET]: "0x3bA9DF642f5e895DC76d3Aa9e4CE8291108E65b1",
+  [ChainId.PGN]: "0x03506eD3f57892C85DB20C36846e9c808aFe9ef4",
+  [ChainId.PGN_TESTNET]: "0x4268900E904aD87903De593AA5424406066d9ea2",
+  [ChainId.FANTOM_MAINNET_CHAIN_ID]:
+    "0x03506eD3f57892C85DB20C36846e9c808aFe9ef4",
+  [ChainId.OPTIMISM_MAINNET_CHAIN_ID]:
+    "0x15fa08599EB017F89c1712d0Fe76138899FdB9db",
+  [ChainId.FANTOM_TESTNET_CHAIN_ID]:
+    "0x0x62a850d7805f3Ae382C6eEf7eEB89A31f68Ce2d5",
```
Comment (gravityblast) [inline packages/common/src/allo/addresses/mrc.ts:16]: there's a double `0x` prefix

Comment (vacekj) [reply packages/common/src/allo/addresses/mrc.ts:16]: good catch, thanks!

**Comment by gravityblast** [comment]
it's still failing to send the tx at checkout

**Comment by gravityblast** [comment]
the current step is no longer animated (the blue one)

**Comment by gravityblast** [comment]
I still have the same issue following the steps above

File: packages/grant-explorer/src/features/round/ViewCartPage/MRCProgressModalBody.tsx [hunk 2]
```diff
@@ -173,7 +173,7 @@ export function MRCProgressModalBody({
                   step={step}
                   icon={
                     <span
-                      className="relative z-10 w-6 h-6 flex items-center justify-center bg-teal-500 rounded-full"
+                      className="relative z-10 w-6 h-6 flex items-center justify-center bg-teal-500 rounded-full "
```
Comment (gravityblast) [inline packages/grant-explorer/src/features/round/ViewCartPage/MRCProgressModalBody.tsx:176]: there's a space added here

**Review by michellema1208 — CHANGES_REQUESTED** [review]
@vacekj - I tested just now and can confirm that donations on Sepolia are working (following @gravityblast's instructions above) and I also additionally tested MRC with a Sepolia + Optimism project in cart - also worked.

However, noted one UX quirk that seems to be introduced by this PR as its not happening in prod -- when you donate to two or more rounds, in the horizontal progress bar in the checkout modal, when the first step is completed, the check continues to flash and appears to enlarge. Whereas in prod, the check remains constant. 

This PR: 
Screen Shot 2024-02-23 at 4 33 34 PM

Sepolia round: https://explorer-staging-git-2817-explorer-move-v1-d7e21b-grants-stack.vercel.app/#/round/11155111/0x0a651f9aba8be2280557013d0bcdfeeb44294b19 
Optimism round: https://explorer-staging-git-2817-explorer-move-v1-d7e21b-grants-stack.vercel.app/#/round/10/0xd52b1ecf94dcc8036180ca2e6b63e05d78bc4cde 

Prod: 
Screen Shot 2024-02-23 at 4 35 43 PM

Prod Optimism round: https://explorer.gitcoin.co/#/round/10/0xd52b1ecf94dcc8036180ca2e6b63e05d78bc4cde
Prod Base round: https://explorer.gitcoin.co/#/round/8453/0xb84909f60b31db387baea1ea8c6dfe1b16100eee

I know it's super small, but want to callout even subtle UX regressions early so we don't create more debt for ourselves in the future. Feel free to test yourself with the links provided above!

+@melissa-neira to test as well to help callout any other UX/design notes!

**Comment by michellema1208** [comment]
@vacekj just to ask explicitly: is seeing the donations on the Donation History page in scope for this PR?

**Comment by vacekj** [comment]
@michellema1208 donation history page shouldn't be affected by this PR at all - that's an indexer-populated page.

**Review by boudra — CHANGES_REQUESTED** [review]

**Comment by michellema1208** [comment]
> @michellema1208 donation history page shouldn't be affected by this PR at all - that's an indexer-populated page.

@vacekj cool, thank you! Does that mean that donations made within this PR would show up on the Donation History page? Or is additional Indexer work needed in order to support that? +@boudra

**Comment by boudra** [comment]
> > @michellema1208 donation history page shouldn't be affected by this PR at all - that's an indexer-populated page.
> 
> @vacekj cool, thank you! Does that mean that donations made within this PR would show up on the Donation History page? Or is additional Indexer work needed in order to support that? +@boudra

yes they should show up, I think what @vacekj meant to say was that the history page was not modified in this PR therefore it should be working as normal

**Comment by vacekj** [comment]
exactly @michellema1208, the donation history page is not modified in this PR, so any bugs there warrant a new ticket.

**Review by gravityblast — APPROVED** [review]

**Review by boudra — APPROVED** [review]

**Pull request merged**